drkafka/src/main/java/com/pinterest/doctorkafka/KafkaCluster.java

```diff
@@ -223,17 +226,47 @@ public BrokerStats getLatestBrokerStats(int brokerId) {
         new PriorityQueue<>(new KafkaBroker.KafkaBrokerComparator());
     for (Map.Entry<Integer, KafkaBroker> entry : brokers.entrySet()) {
       KafkaBroker broker = entry.getValue();
-      BrokerStats latestStats = broker.getLatestStats();
-      if (latestStats == null
-          || latestStats.getHasFailure()
-          || System.currentTimeMillis() - latestStats.getTimestamp() > 240000) {
+      if (isInvalidBroker(broker)) {
         continue;
       }
       brokerQueue.add(broker);
     }
     return brokerQueue;
   }
 
+  /**
+   *
+   * @return a priority queue of brokers for each locality in the cluster ordered by network stats
+   */
+  public Map<String, PriorityQueue<KafkaBroker>> getBrokerQueueByLocality(){
+    Map<String, PriorityQueue<KafkaBroker>> brokerLocalityMap = new HashMap<>();
+    Comparator<KafkaBroker> comparator = new KafkaBroker.KafkaBrokerComparator();
+    for ( Map.Entry<Integer, KafkaBroker> entry : brokers.entrySet() ){
+      KafkaBroker broker = entry.getValue();
+      if (isInvalidBroker(broker)){
+        continue;
+      }
+      // add broker to locality queue
+      // init queue if queue not present in brokerMap for a locality
+      brokerLocalityMap
+          .computeIfAbsent(broker.getRackId(), i -> new PriorityQueue<>(comparator))
+          .add(broker);
+    }
+    return brokerLocalityMap;
+  }
+
+  /**
+   * checks if the broker is invalid for assigning replicas
+   * @param broker the broker that we want to check
+   * @return true if the broker is invalid for assigning replicas, false if it is valid
+   */
+  protected boolean isInvalidBroker(KafkaBroker broker) {
+    BrokerStats latestStats = broker.getLatestStats();
+    return latestStats== null ||
+        latestStats.getHasFailure() ||
+        System.currentTimeMillis() - latestStats.getTimestamp() > 240000;
```

Comment:
could we define `240000` as a constant at the beginning of the class?   in the previous implementation we did not declare as a static constant. let's take opportunity to fix this issue.